server/src/main/java/org/elasticsearch/action/bulk/Retry.java

```diff
@@ -124,7 +124,14 @@ private void retry(BulkRequest bulkRequestForRetry) {
             assert backoff.hasNext();
             TimeValue next = backoff.next();
             logger.trace("Retry of bulk request scheduled in {} ms.", next.millis());
-            retryCancellable = scheduler.schedule(() -> this.execute(bulkRequestForRetry), next, ThreadPool.Names.SAME);
+            bulkRequestForRetry.incRef();
```

Comment:
I initially thought this was inefficient relative to what we currently do but:

Retrying, will cause the whole request to be retained instead of just those requests that actually get retried. But, I think in the real world this is less of an issue with where we currently are.
If the retry happens on a coordinating node, we already have this problem (since we use shared bytes that are tracked as one thing for the whole bulk request). If the retry happens for a replication request, then the replication request is the same unit of work as the primary bulk request received so that seems fine as well. There's certainly room here for more selectively retaining bytes though but I don't think this PR makes us worse off here.